### ISSUE TITLE
Unable to run solution - The object is used in the context different from the one associated with the object.

### ISSUE THREAD

**Issue by julesdadrummer** [Barbarur/NovaPoint#16]
2024-05-29 15:34:31 - [NPLogger] - Solution has started, please wait to the end
2024-05-29 15:34:35 - [AppInfo] - Finish aquiring new Access Token from AAD
2024-05-29 15:34:35 - [AppInfo] - Access Token expiration time: 2024-05-29 4:56:38 PM +00:00
2024-05-29 15:34:35 - [GraphAPIHandler] - Request Message Uir https://graph.microsoft.com/v1.0/sites/root
2024-05-29 15:34:36 - [GraphAPIHandler] - Successful response
2024-05-29 15:34:36 - [Appinfo] - Hostname: redacted.sharepoint.com
2024-05-29 15:34:36 - [Appinfo] - Domain: redacted
2024-05-29 15:34:36 - [AppInfo] - SPO Admin URL 'https://redacted-admin.sharepoint.com'
2024-05-29 15:34:36 - [AppInfo] - SPO -my URL 'https://redacted-my.sharepoint.com'
2024-05-29 15:34:36 - [AppInfo] - SPO root URL 'https://redacted.sharepoint.com'
2024-05-29 15:34:36 - [GetAADUser] - Getting Signed-in user
2024-05-29 15:34:36 - [AppInfo] - Got access token from memory
2024-05-29 15:34:36 - [AppInfo] - Access Token expiration time: 2024-05-29 4:56:38 PM +00:00
2024-05-29 15:34:36 - [GraphAPIHandler] - Request Message Uir https://graph.microsoft.com/v1.0/me
2024-05-29 15:34:36 - [GraphAPIHandler] - Successful response
2024-05-29 15:34:36 - [AppInfo] - Getting Access Token for root site https://redacted-admin.sharepoint.com
2024-05-29 15:34:36 - [AppInfo] - Finish aquiring new Access Token with Refresh Token.
2024-05-29 15:34:36 - [AppInfo] - Access Token expiration time: 2024-05-29 4:52:42 PM +00:00
2024-05-29 15:34:36 - [SPOSiteCollectionCSOM] - Getting Site Collections; IncludePersonalSite 'False', Group ID Defined 'False'
2024-05-29 15:34:41 - [SPOSiteCollectionCSOM] - Collected 1430 Site Collections...
2024-05-29 15:34:41 - [AppInfo] - Getting Access Token for root site https://redacted-admin.sharepoint.com
2024-05-29 15:34:41 - [AppInfo] - Got access token from memory
2024-05-29 15:34:41 - [AppInfo] - Access Token expiration time: 2024-05-29 4:52:42 PM +00:00
2024-05-29 15:34:41 - [NPLogger] - The object is used in the context different from the one associated with the object.

**Comment by Barbarur** [comment]
Which Solution are you using?

Please open the log file and share the top part, so I can see the parameters used for the solution.

Can you share also the message trace from the error message?

**Comment by julesdadrummer** [comment]
![2024-05-30 07_50_07-PermissionsReport_20240529153431_Logs txt - Notepad](https://github.com/Barbarur/NovaPoint/assets/61877493/42639f1e-ff23-43fe-b925-2fe321c7a092)

**Comment by julesdadrummer** [comment]
Not sure how to get the message trace, let me know and I'll dig into it. No matter what settings I use, I get the same object is used in the context error. (All related to the User Permissions Solution)

Thank you!

**Comment by Barbarur** [comment]
Found the issue. Thank you for reporting. I'll fix it soon.

Meanwhile you can run the report for a single site or for a list of sites. Considering you want the report to include "Site Collection Admins", "Site Permissions" and "Unique Permissions". This is very heavy and it can take quite long to finish, you might want to break it down in batches of several sites.

I'll inform here once the fix is released.

**Comment by julesdadrummer** [comment]
Thank you! Would it be possible to also allow specific encoding for the data export? I have user information that needs to be encoded in UTF-8 to properly display the accents etc. Currently it's partially unreadable.
Example: SharePoint Group '4286 _ Accompagnement **Ã**  la mise **Ã**  jour du plan **stratÃ©gique** - Visiteurs'

**Comment by Barbarur** [comment]
Hi @julesdadrummer ,

The issue with special characters is coming from how Excel open CSV files. I personally open CSV files with LibreOffice which encodes the data correctly when opening the file. But when I open it in Excel I got the same issue as you.

I found this article (https://support.microsoft.com/en-us/office/opening-csv-utf-8-files-correctly-in-excel-8a935af5-3416-4edd-ba7e-3dfd2bc4a032), I tested it from my side and Excel encodes the data in UTF-8 when you use *Get data > From file > From text/CSV*.
